Signal/src/ViewControllers/PhotoLibrary/PhotoCollectionPickerController.swift

```diff
@@ -0,0 +1,146 @@
+
+    private weak var collectionDelegate: PhotoCollectionPickerDelegate?
+
+    private let library: PhotoLibrary
+    private let lastPhotoCollection: PhotoCollection
+    private var photoCollections: PhotoCollections
+
+    required init(library: PhotoLibrary,
+                  lastPhotoCollection: PhotoCollection,
+                  collectionDelegate: PhotoCollectionPickerDelegate) {
+        self.library = library
+        self.lastPhotoCollection = lastPhotoCollection
+        self.photoCollections = library.allPhotoCollections()
+        self.collectionDelegate = collectionDelegate
+        super.init()
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: View Lifecycle
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        let titleLabel = UILabel()
+        titleLabel.text = lastPhotoCollection.localizedTitle()
+        titleLabel.textColor = Theme.primaryColor
+        titleLabel.font = UIFont.ows_dynamicTypeBody.ows_mediumWeight()
+
+        let titleIconView = UIImageView()
+        titleIconView.tintColor = Theme.primaryColor
+        titleIconView.image = UIImage(named: "navbar_disclosure_up")?.withRenderingMode(.alwaysTemplate)
+
+        let titleView = UIStackView(arrangedSubviews: [titleLabel, titleIconView])
+        titleView.axis = .horizontal
+        titleView.alignment = .center
+        titleView.spacing = 5
+        titleView.isUserInteractionEnabled = true
+        titleView.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(titleTapped)))
+        navigationItem.titleView = titleView
+
+        library.add(delegate: self)
+
+        navigationItem.leftBarButtonItem = UIBarButtonItem(barButtonSystemItem: .cancel,
+                                                           target: self,
+                                                           action: #selector(didPressCancel))
+
+        updateContents()
+    }
+
+    private func updateContents() {
+        photoCollections = library.allPhotoCollections()
+
+        let section = OWSTableSection()
+        let count = photoCollections.count
+        for index in 0..<count {
+            let collection = photoCollections.collection(at: index)
+            section.add(OWSTableItem.init(customCellBlock: { () -> UITableViewCell in
+                let cell = OWSTableItem.newCell()
+
+                let imageView = UIImageView()
+                let kImageSize = 50
+                imageView.autoSetDimensions(to: CGSize(width: kImageSize, height: kImageSize))
+
+                let contents = collection.contents()
+                if contents.count > 0 {
+                    let photoMediaSize = PhotoMediaSize(thumbnailSize: CGSize(width: kImageSize, height: kImageSize))
+                    let assetItem = contents.assetItem(at: 0, photoMediaSize: photoMediaSize)
```

Comment:
Is it possible to have an empty album? Will this explode in that case?

Reply:
There's a `contents.count` condition two lines up.